### DISCUSSION TITLE
adding gef-scripts

### DISCUSSION THREAD

**Discussion by hros** [hugsy/gef#924] [General]
I am following the introductory video: [Advanced exploitation with GDB-GEF: GEF 101 - Solving pwnable.tw/start](https://www.youtube.com/watch?v=KWG7prhH-ks)
at approximately 13:30 in the video there is an explanation how to add the gef-scripts:
* `gef‣ !git clone https://github.com/hugsy/gef-scripts`
* `gef‣ gef config gef.extra_plugins_dir "~/gef-scripts"` 
* `gef‣ gef save` 
* `gef‣ quit` 
after this sequence, relaunching gdb `gdb -q ./start` (start is the name of the executable being debugged) returns the error:
> warning: /home/user/gef-scripts/.github/__init__.py: No such file or directory
> warning: /home/user/gef-scripts/structs/__init__.py: No such file or directory
> warning: /home/user/gef-scripts/docs/__init__.py: No such file or directory
> warning: /home/user/gef-scripts/archs/__init__.py: No such file or directory
> warning: /home/user/gef-scripts/.git/__init__.py: No such file or directory
>warning: /home/user/gef-scripts/os/__init__.py: No such file or directory

If I change the directory line in `.gef.rc` 
from: `extra_plugins_dir = ~/gef-scripts` 
to: `extra_plugins_dir = ~/gef-scripts/scripts`
I get the following:
```
GEF for linux ready, type `gef' to start, `gef config' to configure
90 commands loaded and 5 functions added for GDB 12.1 in 0.00ms using Python engine 3.10
Traceback (most recent call last):
  File "/home/user/gef-scripts/scripts/libc_function_args/__init__.py", line 139, in <module>
    register_external_context_pane(
TypeError: register_external_context_pane() got an unexpected keyword argument 'condition'
Traceback (most recent call last):
  File "/home/user/gef-scripts/scripts/emulate/__init__.py", line 11, in <module>
    import capstone
  File "/home/user/gef-scripts/scripts/capstone.py", line 16, in <module>
    __cs: Optional[capstone.Cs] = None
AttributeError: partially initialized module 'capstone' has no attribute 'Cs' (most likely due to a circular import)
Traceback (most recent call last):
  File "/home/user/gef-scripts/scripts/syscall_args/__init__.py", line 138, in <module>
    register_external_context_pane(
TypeError: register_external_context_pane() got an unexpected keyword argument 'condition'
Traceback (most recent call last):
  File "/home/user/gef-scripts/scripts/ropper.py", line 8, in <module>
    import ropper
  File "/home/user/gef-scripts/scripts/ropper.py", line 11, in <module>
    @register
NameError: name 'register' is not defined
Traceback (most recent call last):
  File "/home/user/gef-scripts/scripts/capstone.py", line 9, in <module>
    import capstone
  File "/home/user/gef-scripts/scripts/capstone.py", line 16, in <module>
    __cs: Optional[capstone.Cs] = None
AttributeError: partially initialized module 'capstone' has no attribute 'Cs' (most likely due to a circular import)
Traceback (most recent call last):
  File "/home/user/gef-scripts/scripts/assemble.py", line 9, in <module>
    import keystone
ModuleNotFoundError: No module named 'keystone'
[+] 36 extra commands added from '~/gef-scripts/scripts' in 0.03 seconds
Reading symbols from ./start...
(No debugging symbols found in ./start)
```

**Comment by clark-ja** [comment]
Not sure this qualifies as a "Discussion" but here we go.

First of all it is called `gef-extras` now. The Github repository is located at https://github.com/hugsy/gef-extras. Please do not follow installation instructions from a video which has been released 4 years ago.

You can find the documentation for gef-extras [here](https://hugsy.github.io/gef-extras/) and on [this page](https://hugsy.github.io/gef-extras/install/) you can find all the installation instructions for it. If you want to go for a similar way of installation as the one you posted above please read the "Installation using Git" section AND the "Python dependencies" section as I can see in your output from above that you e.g. don't have the python packages keystone, capstone, ropper installed.

Please let us know if going through the recent version of the documentation helped you.

**Reply by hros** [reply]
thanks for the thorough instructions
I was able to add `gef-extras`
I still haven't tried out any of the functions, I hope they work. I suggest adding a command or commands in the installation instructions to verify that everything works (and that all dependencies are met)
btw, since browsers add a copy button to the code snippet frames it would be beneficial to leave out the prefix ($, gef>, ...) which gets copied

thanks again

**Reply by hugsy** [reply]
@hros 

For installation issues, the best and fastest way is Discord. Issues are for well, issues (i.e. real bugs) and Discussions for improvements or topics about that don't qualify as Issues. Setup issues are neither.

Glad you fixed your problem.
